Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptTypeEditController.java

```diff
@@ -27,84 +33,98 @@
-		types.remove(typeid);
-		model.addAttribute("supertypes", types);
-
-		return "/auth/concepttype/edittype";
-	}
-
-	/**
-	 * This method stores the updated information of a type
-	 * 
-	 * @param typeid
-	 * @param req
-	 *            holds HTTP request information
-	 * @param principal
-	 *            holds logged in user information
-	 * @return Returns a string value to redirect user to type list page
-	 */
-	@RequestMapping(value = "auth/concepttype/storeedittype/{typeid}", method = RequestMethod.POST)
-	public String editType(@PathVariable("typeid") String typeid,
-			HttpServletRequest req, Principal principal) {
-
-		ConceptType type = typeManager.getType(typeid);
-		type.setTypeName(req.getParameter("name"));
-		type.setDescription(req.getParameter("description"));
-		type.setMatches(req.getParameter("match"));
-		type.setSupertypeId(req.getParameter("supertypes"));
-
-		String userId = usersManager.findUser(principal.getName()).getUsername();
-		String modified = type.getModified() != null ? type.getModified() : "";
-		if (!modified.trim().isEmpty())
-			modified += ", ";
-		type.setModified(modified + userId + "@" + (new Date()).toString());
-
-		typeManager.storeModifiedConceptType(type);
-		return "redirect:/auth/concepttype";
-	}
-
-	/**
-	 * This method returns the string to redirect user to type list page when
-	 * user cancels type edit operation
-	 * 
-	 * @return Returns a string value to redirect user to concept type list page
-	 */
-	@RequestMapping(value = "auth/concepttype/edittype/canceledit", method = RequestMethod.GET)
-	public String cancelEdit() {
-		return "redirect:/auth/concepttype";
-	}
+    @Autowired
+    private IConceptTypeManger typeManager;
+
+    @Autowired
+    private IUserManager usersManager;
+
+    @Autowired
+    private ConceptTypeAddValidator validator;
+
+    @InitBinder
+    protected void initBinder(WebDataBinder binder) {
+        binder.setValidator(validator);
+    }
+
+    /**
+     * This method provides a type information to edit type page
+     * 
+     * @param typeid
+     *            Represents a type which has to be edited
+     * @param model
+     *            Generic model holder for servlet
+     * @return Returns a string value to redirect user to type edit page
+     */
+    @RequestMapping(value = "auth/concepttype/edittype/{typeid}", method = RequestMethod.GET)
+    public String prepareEditType(@PathVariable("typeid") String typeid, ModelMap model,
+            @ModelAttribute("conceptTypeAddForm") ConceptTypeAddForm conceptTypeAddForm) {
+
+        ConceptType type = typeManager.getType(typeid);
+
+        conceptTypeAddForm.setTypeName(type.getTypeName());
+        conceptTypeAddForm.setOldTypeName(type.getTypeName());
+        conceptTypeAddForm.setTypeDescription(type.getDescription());
+        conceptTypeAddForm.setMatches(type.getMatches());
+        conceptTypeAddForm.setSelectedType(type.getSupertypeId().trim());

```

Comment:
My question here is, why it is "selectedType"? It should be supertype as it is the same, right? Makes it very confusing if you change the variable name from supertype to selected type, or is there a reason it is called "selected type" and not "supertype"?